### ISSUE TITLE
[Question] Should the VM create command create a new NSG when referencing an existing NIC?

### ISSUE THREAD

**Issue by lostintangent** [Azure/azure-cli#1659]
If I call `az vm create`, and specify an existing NIC via the `--nics` flag, I notice that I still need to explicitly set the `--nsg ''` parameter if I want to prevent it from creating a new NSG.

This behavior seemed a little strange to me since [these docs](https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-nsg) seem to indicate that the NSG is a NIC-associated resource. Therefore, if a user is explicitly managing the NIC, as opposed to relying on the `az vm create` uber command to create it (along with other resources), then it seemingly shouldn't create another NSG, under the assumption that they intend to manage the NSG as well. That may be the incorrect way to think about this, so I'm just thinking out loud here.

Calling `az vm create` with the `--nics` flag seems to suppress it from creating all other resources (e.g. public IP), so the NSG seems to be the only thing that requires being explicitly turned off. Ideally, I could call the following:

```shell
az vm create -g foo-group -n foo --image UbuntuLTS --nics foo-nic
```

As opposed to this:

```shell
az vm create -g foo-group -n foo --image UbuntuLTS --nics foo-nic --nsg ''
```

**Comment by tjprescott** [comment]
@lostintangent great point! Pretty much, if you specify NICs, you should not supply any of the parameters for NSG, VNET or PublicIP. We can fix that defaulting logic. Thanks for pointing this out!

**Comment by lostintangent** [comment]
@tjprescott Awesome! It's such a small thing, but it will "feel" so much nicer to not need to add that `--nsg ''` flag :)

**Comment by tjprescott** [comment]
Fixed by #1849
